internal/host/transition_test.go

```diff
@@ -6708,6 +6709,19 @@ var _ = Describe("State machine test - refresh transition", func() {
 			Expect(stateMachine.Run(TransitionTypeRefresh, testState, &refreshHostArgs)).To(Succeed())
 			Expect(string(testState.State())).To(Equal(models.HostStatusKnown))
 		})
+
+		It("Moves from known to insufficient when iscsi-host-network-interface-does-not-belong-to-machine-cidr validation fails", func() {
+
+			refreshHostArgs.conditions[string(IscsiHostNetworkInterfaceDoesNotBelongToMachineCidr)] = false
+
+			Expect(stateMachine.Run(TransitionTypeRefresh, testState, &refreshHostArgs)).To(Succeed())
+			Expect(string(testState.State())).To(Equal(models.HostStatusInsufficient))
+
+			refreshHostArgs.conditions[string(IscsiHostNetworkInterfaceDoesNotBelongToMachineCidr)] = true
+
+			Expect(stateMachine.Run(TransitionTypeRefresh, testState, &refreshHostArgs)).To(Succeed())
+			Expect(string(testState.State())).To(Equal(models.HostStatusKnown))
```

Comment:
same as before let make it a practice to add debug message to Expect .
This will help see the root cause and not some generic failure

Reply:
I think it's a great idea to add error expectations, but I think it should be done by type and not by error/debug message. Otherwise tests will be even more fragile (text change will make us fix tests) , and our codebase is already struggling with test fragility.

Reply:
let's take it offline